lib/pixelutil.ex

```diff
@@ -3,51 +3,57 @@ defmodule Inky.PixelUtil do
   `PixelUtil` maps pixels to bitstrings to be sent to an Inky screen
   """
 
-  def pixels_to_bitstring(pixels, display, color) do
-    rotation = display.rotation / 90
-    width = display.width
-    height = display.height
-
-    p2b =
-      case color do
-        :black -> &pixel_to_black/1
-        _ -> &pixel_to_accent/1
-      end
-
-    opts = bitstring_traversal_opts(rotation, width, height)
-    do_pixels_to_bitstring(pixels, p2b, opts)
+  def pixels_to_bits(pixels, width, height, rotation_degrees, color_map) do
+    opts = bitstring_traversal_opts(rotation_degrees, width, height)
+    do_pixels_to_bits(pixels, color_map, opts)
   end
 
-  defp pixel_to_black(:black), do: 0
-  defp pixel_to_black(_), do: 1
+  @doc """
+  Only exposed for testing purposes. Do not use.
 
-  defp pixel_to_accent(c) when c in [:red, :yellow, :accent], do: 1
-  defp pixel_to_accent(_), do: 0
+  ## Doctests
+
+      iex> Inky.PixelUtil.bitstring_traversal_opts(0, 212, 104)
+      {:y_outer, 0, 103, 0, 211}
+
+      iex> Inky.PixelUtil.bitstring_traversal_opts(180, 212, 104)
+      {:y_outer, 103, 0, 211, 0}
+
+      iex> Inky.PixelUtil.bitstring_traversal_opts(-90, 212, 104)
+      {:x_outer, 211, 0, 0, 103}
+
+      iex> Inky.PixelUtil.bitstring_traversal_opts(90, 212, 104)
+      {:x_outer, 0, 211, 103, 0}
+
+  """
+  def bitstring_traversal_opts(rotation_degrees, width, height) do
+    # Simplify and wrap around rotations
+    rotation =
+      rotation_degrees
+      |> div(90)
+      |> rem(4)
+      |> (&if(&1 < 0, do: &1 + 4, else: &1)).()
 
-  defp bitstring_traversal_opts(rotation, width, height) do
     w_n = width - 1
     h_n = height - 1
 
     case rotation do
-      -1.0 -> {:x_outer, w_n, 0, 0, h_n}
-      1.0 -> {:x_outer, 0, w_n, h_n, 0}
-      -2.0 -> {:y_outer, h_n, 0, w_n, 0}
-      _ -> {:y_outer, 0, h_n, 0, w_n}
+      3 -> {:x_outer, w_n, 0, 0, h_n}
+      1 -> {:x_outer, 0, w_n, h_n, 0}
+      2 -> {:y_outer, h_n, 0, w_n, 0}
+      0 -> {:y_outer, 0, h_n, 0, w_n}
```

Comment:
Results should be defined like:

``` elixir
@type outer_loop :: :x | :y
@type x_direction :: :x_inc | :x_dec
@type y_direction :: :y_inc | :y_dec
@type axis_directions :: {x_direction(), y_direction} | {y_direction(), x_direction()}
@type result :: {outer_loop(), axis_directions()}
```

It would make the code less confusing by having named values instead of this number/variable soup.